### DISCUSSION TITLE
Possible to use ReShade?

### DISCUSSION THREAD

**Discussion by Javic36** [diasurgical/DevilutionX#5503] [General]
I've always found Diablo's palette a bit subdued. Can [ReShade ](https://reshade.me/) be used with this EXE? I tried all three forms (DX9, DX10/11, and OpenGL) and none work. Perhaps there's a special way to install it for this game?

**Comment by qndel** [comment]
Since reshade looks to be windows only I doubt we'd put any effort into that.

**Reply by AJenbo** [reply]
When using upscaling DevilutionX will use either Dx9/DX11/OpenGL depending on your graphics card. If upscaling is disabled it will instead output directly to the window using a software path which probably won't work with ReShade.

This might also be relevant for you: https://github.com/diasurgical/devilutionX/issues/4452

**Comment by Javic36** [comment]
Thanks to both responders. I do have upscaling enabled but still no joy.

Yes, ReShade is Windows-only. As I understand it though no effort is required on the part of game developers. It replaces the relevant rendering DLL (whether DirectX or OpenGL) with one that injects custom shaders. The game remains ignorant of the deception. I don't know enough about the technology to guess why it doesn't work with Devilution. (Maybe I'll pose that question to the ReShade support forum.)

That other post  (#4452) mentions LibRetro and cnc-ddraw, neither of which I was familiar with (I'm no expert on this topic), but both sound like they enable (or at least allow) postprocessing shader support. Devilution Issue #4453 requests LibRetro shader support, but there's no info about if/when that feature will be available.

Perhaps I should have asked a more general question: Is some form of postprocessing shader support available? If so, how can I use it? If not, is it planned for a future release?

Not a big issue--Devilution is still great for playing Diablo in 2022--but a little extra color would still be nice (especially in winter :-).

**Reply by AJenbo** [reply]
> Yes, ReShade is Windows-only. As I understand it though no effort is required on the part of game developers.

If that is the case then there I don't see what we can do. I'm not really familiar with ReShader or even Windows.

I think talking to the ReShader developers is the right choice here. If nothing is required of us then it's not working as advertised.

It does look like others have been able to get it working:
https://www.youtube.com/watch?v=icfbOhFO8wM

> Devilution Issue #4453 requests LibRetro shader support, but there's no info about if/when that feature will be available.

We don't work on a time schedule this is just a hobby project.

> Is some form of postprocessing shader support available?

Not explicitly, but there is also nothing specifically preventing it.

> If not, is it planned for a future release?

See #4453

> a little extra color would still be nice (especially in winter :-).

Maybe check monitor settings or the display color profile in Windows

**Reply by Javic36** [reply]
Thanks! I did post to the ReShade Discord but got a not-very-helpful reply. That YouTube video is _very_ interesting. I'll try contacting that person to see if they can help me. (If I get it working, I'll post back here in case others are interested.) Thanks again.

**Reply by qndel** [reply]
Notice the youtube video is 3 years old. I think at that point DevilutionX wasn't very different from Devilution.

**Reply by AJenbo** [reply]
You could try disabling upscaling and see if it's actually able to work with that

**Comment by Javic36** [comment]
Inspired by the video, I played around a bit more and got it working. The "fix" is to run the 32-bit Devilution. I was originally running the 64-bit version because I have a 64-bit OS. When I switched to the 32-bit, ReShade started working exactly as expected. Cool!

**Reply by AJenbo** [reply]
That's probably because you are using a 32bit ReShade you need a 64bit one for it to work with 64bit applications

**Reply by Javic36** [reply]
Y'know, I don't think I ever realized that ReShade came in both 32-bit and 64-bit versions even after using it for years. :-)  I had installed it "manually" and got the 32-bit binaries. But when I ran the actual installer (like you're supposed to do :-/), it installed a version that works fine with 64-bit Devilution. So "user error" on my part. Sorry for the trouble. (Embarrasing)
